services/file-retriever/__tests__/partialDownloads.spec.ts

```diff
@@ -23,93 +21,41 @@ describe('Partial downloads', () => {
   })
 
   it('should return the correct chunk', async () => {
-    const nodeMapByCID: Record<string, PBNode> = {}
-    const chunks = ['a', 'b', 'c', 'd', 'e']
-
-    // Create the tree of nodes
-    const nodes = chunks.map((chunk) =>
-      createFileChunkIpldNode(Buffer.from(chunk, 'utf-8')),
+    const chunks = [
+      {
+        cid: 'bafkr6idz7htrqhks6xyntulrqfyevx3k5qrhptbmgoxlmbss65yi3u25ym',
+        links: [],
+        type: MetadataType.File,
+        linkDepth: 0,
+        blockHeight: 0,
+        blockHash: '',
+        extrinsicId: '',
+        extrinsicHash: '',
+        indexInBlock: 0,
+        blake3Hash: '',
+        timestamp: new Date(),
+      },
+    ]
+    jest.spyOn(dsnFetcher, 'getFileChunks').mockResolvedValue(chunks)
+
+    const fetchNodeSpy = jest
+      .spyOn(dsnFetcher, 'fetchNode')
+      .mockImplementation(async (cid) => ({
+        Data: IPLDNodeData.encode({
+          type: MetadataType.File,
+          data: Buffer.from(cid),
+        }),
+        Links: [],
+      }))
+
+    const chunk = await dsnFetcher.getPartial(
+      'bafkr6idz7htrqhks6xyntulrqfyevx3k5qrhptbmgoxlmbss65yi3u25ym',
+      0,
     )
-    const headNode = createChunkedFileIpldNode(
-      nodes.map((node) => cidOfNode(node)),
-      5n,
-      1,
+    expect(chunk).toEqual(Buffer.from(chunks[0].cid))
+    expect(fetchNodeSpy).toHaveBeenCalledWith(
+      'bafkr6idz7htrqhks6xyntulrqfyevx3k5qrhptbmgoxlmbss65yi3u25ym',
+      chunks.map((e) => e.cid),
     )
-
-    // Create a map of the nodes by CID
-    nodes.forEach((node) => {
-      const cid = cidToString(cidOfNode(node))
-      nodeMapByCID[cid] = node
-    })
-    nodeMapByCID[cidToString(cidOfNode(headNode))] = headNode
-
-    const cid = cidToString(cidOfNode(headNode))
-
-    jest.spyOn(dsnFetcher, 'fetchNode').mockImplementation(async (cid) => {
-      return nodeMapByCID[cid]
-    })
-
-    // First node (head) should return null
-    const chunk = await dsnFetcher.getPartial(cid, 0)
-    expect(chunk).toBeInstanceOf(Buffer)
-    expect(chunk?.length).toBe(0)
-
-    // Chunks 1-5 should return the correct chunk
-    let i = 0
-    while (i < chunks.length) {
-      // i + 1 for offsetting the head node
-      const chunk = await dsnFetcher.getPartial(cid, i + 1)
-      expect(chunk).toBeDefined()
-      expect(chunk?.toString('utf-8')).toBe(chunks[i])
-      i++
-    }
-
-    // Chunk 6 should return empty buffer because it's past the end of the file
-    const chunk6 = await dsnFetcher.getPartial(cid, 6)
-    expect(chunk6).toBeNull()
-  })
-
-  it('should work with a large file', async () => {
```

Comment:
Large file tess doesn't make much sense anymore since the IPLD tree is already flatten